dotnet_sdk/lib/dependabot/dotnet_sdk/file_updater.rb

```diff
@@ -0,0 +1,60 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "dependabot/file_updaters"
+require "dependabot/file_updaters/base"
+require "sorbet-runtime"
+
+module Dependabot
+  module DotnetSdk
+    class FileUpdater < Dependabot::FileUpdaters::Base
+      extend T::Sig
+
+      sig { override.returns(T::Array[Regexp]) }
+      def self.updated_files_regex
+        [/^global\.json$/]
+      end
+
+      sig { override.returns(T::Array[Dependabot::DependencyFile]) }
+      def updated_dependency_files
+        updated_files = []
+
+        contents = update
+
+        updated_files << updated_file(file: global_json, content: contents) if file_changed?(global_json)
+
+        updated_files
+      end
+
+      private
+
+      sig { returns(Dependabot::Dependency) }
+      def dependency
+        # Dockerfiles will only ever be updating a single dependency
+        T.must(dependencies.first)
+      end
+
+      sig { override.void }
+      def check_required_files
+        # Just check if there are any files at all.
+        return if dependency_files.any?
+
+        raise "No global.json configuration!"
+      end
+
+      sig { returns(Dependabot::DependencyFile) }
+      def global_json
+        T.must(dependency_files.find { |f| f.name == "global.json" })
+      end
+
+      sig { returns(String) }
+      def update
+        previous_json = JSON.parse(T.must(global_json.content))
+        previous_json["sdk"]["version"] = T.must(dependency.requirements.first)[:requirement]
+        JSON.pretty_generate(previous_json)
```

Comment:
Not necessary for this PR, but something for the future: many `global.json` files in the wild have comments and custom whitespace formatting (indentation, alignment, etc.) that people like to be preserved.  This is a hard problem and will need lots of iteration to get right.

Reply:
I wonder if we could just do a find + replace for the previous version?

Reply:
That's what I did I my action 😅

Reply:
@martincostello Using this approach actually solves your other issue about updating `tools.dotnet` in lockstep!